prometheus/src/test/scala/filodb/prometheus/parse/ParserSpec.scala

```diff
@@ -726,6 +722,30 @@ class ParserSpec extends AnyFunSpec with Matchers {
     Parser.queryToLogicalPlan(q, qts, step)
   }
 
+  it("should correctly build a label map") {
+    val queryMapPairs = Seq(
+      ("foo", Map("__name__" -> "foo")),
+      ("""{__name__="foo"}""" -> Map("__name__" -> "foo")),
+      ("""{__name__="foo", bar="baz"}""" -> Map("__name__" -> "foo", "bar" -> "baz")),
+      ("""{bar="baz"}""" -> Map("bar" -> "baz")),
+      ("""foo{bar="baz", bog="bah"}""" -> Map("__name__" -> "foo", "bar" -> "baz", "bog" -> "bah")),
+    )
+
+    val queriesShouldFail = Seq(
```

Comment:
`foo{__name__="foo"}` will also fail, but I probably shouldn't explicitly enforce that here.